csharp/ql/src/semmle/code/csharp/frameworks/system/runtime/CompilerServices.qll

```diff
@@ -28,3 +28,38 @@ class SystemRuntimeCompilerServicesTaskAwaiterStruct extends SystemRuntimeCompil
   /** Gets the field that stores the underlying task. */
   Field getUnderlyingTaskField() { result = this.getAField() and result.hasName("m_task") }
 }
+
+/** The `System.Runtime.CompilerServices.ConfiguredTaskAwaitable<>` struct. */
+class SystemRuntimeCompilerServicesConfiguredTaskAwaitableTStruct extends SystemRuntimeCompilerServicesNamespaceUnboundGenericStruct {
+  SystemRuntimeCompilerServicesConfiguredTaskAwaitableTStruct() {
+    this.hasName("ConfiguredTaskAwaitable<>")
+  }
+
+  /** Gets the `GetAwaiter` method. */
+  Method getGetAwaiterMethod() { result = this.getAMethod("GetAwaiter") }
+
+  /** Gets the field that stores the underlying awaiter. */
+  Field getUnderlyingAwaiterField() {
+    result = this.getAField() and result.hasName("m_configuredTaskAwaiter")
+  }
+}
+
+/** An unbound generic struct in the `System.Runtime.CompilerServices` namespace. */
```

Comment:
This comment is not correct. I think this class is so special that there is no need to have it, instead inline the logic where it is used below.